src/Akeneo/Tool/Bundle/MessengerBundle/Command/PurgeMessengerCommand.php

```diff
@@ -0,0 +1,83 @@
+<?php
+declare(strict_types=1);
+
+namespace Akeneo\Tool\Bundle\MessengerBundle\Command;
+
+use Akeneo\Tool\Bundle\MessengerBundle\Query\PurgeDoctrineQueueQuery;
+use Doctrine\DBAL\DBALException;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+/**
+ * @author    Willy Mesnage <willy.mesnage@akeneo.com>
+ * @copyright 2021 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+class PurgeMessengerCommand extends Command
+{
+    protected static $defaultName = 'akeneo:messenger:purge';
+
+    private PurgeDoctrineQueueQuery $purgeDoctrineQueue;
+
+    public function __construct(PurgeDoctrineQueueQuery $purgeDoctrineQueue)
+    {
+        parent::__construct();
+
+        $this->purgeDoctrineQueue = $purgeDoctrineQueue;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function configure()
+    {
+        $this
+            ->setDescription(
+                'Purges the messenger SQL table in terms of the given retention time (default is 120 minutes)'
+            )
+            ->addArgument(
+                'table-name',
+                InputArgument::REQUIRED,
+                'Name of the messenger table to purge.'
+            )
+            ->addArgument(
+                'queue-name',
+                InputArgument::OPTIONAL,
+                'Name of the messenger queue to purge.',
+                'default'
+            )
+            ->addArgument(
+                'retention-time',
+                InputArgument::OPTIONAL,
+                'Deletes messages that are older than the given retention time in seconds.',
+                7200
+            );
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): ?int
+    {
+        $retentionTime = $input->getArgument('retention-time');
+        $tableName = $input->getArgument('table-name');
+        $queueName = $input->getArgument('queue-name');
+        $olderThan = $this->computeOlderThanDateTime((int) $retentionTime);
+
+        try {
+            $this->purgeDoctrineQueue->execute($tableName, $queueName, $olderThan);
+        } catch (DBALException $dbalException) {
+            $output->writeln($dbalException->getMessage());
+
+            return -1;
+        }
+
+        return 0;
+    }
+
+    private function computeOlderThanDateTime(int $retentionTime): \DateTimeImmutable
+    {
+        $retentionTimeAgo = (int) (new \DateTimeImmutable('now'))->format('U') - $retentionTime;
```

Comment:
Aren't we encouraged to do 
```suggestion
        $retentionTimeAgo = (int) (new \DateTimeImmutable('now', new \DateTimeZone('UTC')))->format('U') - $retentionTime;
```
?

Reply:
especially if you specify the timezone the following line.